### PR TITLE
Add Horizontal Streamfunction plots + Subpolar North Atlantic projection 

### PR DESCRIPTION
**Pull request by anirban89**
Summary : 
Config option (`climatologyMapBSF`)  for plotting climatological depth integrated transport streamfunction 
- either full depth (Barotropic) or up to a specified depth (task file = `ocean/climatology_map_bsf.py`)
- option to add arrows to contourlines: config option `arrowsOnContour<suffix>`  in `default.cfg`, files updated `shared/plot/climatology_map.py`, `shared/plot/colormap.py`
Subpolar North Atlantic projection to visualize sub polar gyre.
- subpolar North Atlantic projection added to `shared/projection/__init__.py`, `shared/climatology/comparison_descriptors.py` and `shared/climatology/remap_mpas_climatology_subtask.py` 
- doctoring updated in `shared/climatology/comparison_descriptors.py`

### CODE REVIEW

**Comment by anirban89** [comment]
Hi, Can you all please review this pull request? Maybe test this functionality with other E3SM runs. I only tested with a Southern Ocean refined run: `/lcrc/group/e3sm/ac.abarthel/E3SMv2/20220808.v2.cryo.Bcase.submesoscale.ne30pg2_SOwISC12to60E2r4.chrysalis`.

**Comment by anirban89** [comment]
I changed the figsize of some of the projections (North Pacific, North Atlantic, Antarctic) to help formatting (reduce whitespace etc.) in the default config file, but I only looked at single panel plots and didn't check any of the 3 panel plots. We would need to check how the figures get formatted with the 3 panel plot

**Review by xylar — CHANGES_REQUESTED** [review]
@anirban89, this is really fantastic!  Here are a few comments.  Some are changes I noticed on a quick look through the code.  Some are notes that will need to be followed up on once I've run the full analysis on a typical cryosphere simulation so we can see the effects on our usual plots.

File: mpas_analysis/default.cfg
Comment (xylar) [inline mpas_analysis/default.cfg:1442]: I think this should just be:
```suggestion
comparisonGrids = ['latlon', 'subpolar_north_atlantic']
```

File: mpas_analysis/default.cfg
Comment (xylar) [reply mpas_analysis/default.cfg:388]: Arctic plots will likely need tweaks. See overlapping titles:

image

**Review by xylar — CHANGES_REQUESTED** [review]
A few more comments.

File: mpas_analysis/default.cfg
Comment (anirban89) [reply mpas_analysis/default.cfg:1401]: Yes this was for testing, you are right

File: mpas_analysis/default.cfg
Comment (anirban89) [reply mpas_analysis/default.cfg:1418]: Yes, I only changed the Antarctic, but this was already in need an update before any of my changes no?

File: mpas_analysis/default.cfg
Comment (anirban89) [reply mpas_analysis/default.cfg:1482]: Sorry you are right

File: mpas_analysis/shared/plot/climatology_map.py
Comment (anirban89) [reply mpas_analysis/shared/plot/climatology_map.py:637]: yes, we can remove this for now

**Comment by xylar** [comment]
> Hi, Can you all please review this pull request? 

@anirban89, I think if you don't `@` people, they won't notice this.

@milenaveneziani, @alicebarthel, could you have a look?

We should also ask LeAnn for feedback before we merge.

**Comment by milenaveneziani** [comment]
This looks great! Thanks @anirban89, @xylar. 
I will test this PR on one of the ARRM simulations.

**Comment by milenaveneziani** [comment]
This worked great for me, @anirban89!
Here is the full analysis link: https://portal.nersc.gov/project/m1199/milena/20220810.WCYCL1950.arcticx4v1pg2_ARRM10to60E2r1.baseline_bdvslat/Years11-20/ocean/index.html

And here is the full barotropic SF for the Subpolar gyre:
image

**Comment by xylar** [comment]
@anirban89, this is really fantastic!  I'm going to post some figures where the formatting needs a little work.  Let's talk about how to fix these so we can get this merged.

Here's the full analysis:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis_testing/bsf_vertices2/v2.cryo.Bcase.submesoscale.ne30pg2_SOwISC12to60E2r4.chrysalis/clim_0001-0020_ts_0001-0020/ocean/index.html

Here is a fake main vs. control analysis (with the same run as main and control) so we can look at the 3-panel plots:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis_testing/bsf_main_vs_ctrl/main/clim_0001-0020_ts_0001-0020/ocean/index.html

Too much whitespace between title and panel titles.  Probably, just need to shorten the height of the 3-panel Antarctic figure:
image

Same with Arctic 3-panel:
image

The title font in the Subpolar North Atlantic plots seems a lot bigger than in other plots:
image
This seems to be leading to undesirable whitespace to the left.

We don't plot North Atlantic and North Pacific regions by default but it might be worth doing a test with those regions (both 1 and 3 panel plots) to make sure your modifications here are appropriate.  I'm not super worried about those until we start using them by default, though.

**Comment by xylar** [comment]
@anirban89, it would be good to rebase this branch onto the latest `develop` now that #906 has gone in.  It would be important to make sure the 3 commits from that PR no longer appear here.  We can chat about this tomorrow.

**Comment by anirban89** [comment]
Yes lets reconvene tomorrow

File: mpas_analysis/__main__.py
Comment (xylar) [inline mpas_analysis/__main__.py:149]:
```suggestion
                                            oceanClimatologyTasks['avg'],
```
I fixed this typo in #913 so it should be fixed here, too, once that gets merged and you rebase onto `develop`.

**Comment by anirban89** [comment]
@xylar I am not saying it has to be part of this PR, but I noticed that EKE is being plotted on a linear scale with units in cm^2s^-2, whereas surface speed is plotted in linear scale with units in ms^-1. I have always seen EKE being plotted on a logarithmic scale. Is there any specific reason to not have SI units for this one plot? 
eke_main_ANN_years0001-0020
velocityMagnitudeSOSE_antarctic_main_depth_top_ANN_years0001-0020

**Comment by xylar** [comment]
> @xylar I am not saying it has to be part of this PR, but I noticed that EKE is being plotted on a linear scale with units in cm^2s^-2, whereas surface speed is plotted in linear scale with units in ms^-1. I have always seen EKE being plotted on a logarithmic scale. Is there any specific reason to not have SI units for this one plot? I would expect EKE to be plotted in m^2s^-2 with a logarithmic scale and a different colormap. That will definitely make for a better visualization. Thoughts?

A separate PR to address these things would be very welcome.  The analysis was added by a summer student many years ago.  It hasn't received much attention since then, in part because we don't always even include it in our output.  I agree with all your suggestions: log scale, SI units and a different colormap.  What colormap did you have in mind?  Is there a typical choice that the community expects?

File: mpas_analysis/__main__.py
Comment (xylar) [reply mpas_analysis/__main__.py:149]: @anirban89, this needs to be fixed now.  As soon as it is fixed, I'll re-test the analysis.

Comment (anirban89) [reply mpas_analysis/__main__.py]: I pushed a commit right now

Comment (xylar) [reply mpas_analysis/__main__.py]: Perfect, thanks!

**Comment by anirban89** [comment]
> 

I was thinking something on the lines of this. This is a pretty commonly used colormap for speed and EKE. This is named as Black Blue and White in preview. 
KE 0004

It doesn't appear to match with any of the available maps under MPAS-Tools sciviz color maps.   
colormaps

But I have a python recipe for a custom colormap like this which could work
Unknown-2

**Review by xylar — APPROVED** [review]
@anirban89, this is excellent work!  Further tweaks can be made in follow-up PRs if we find anything that needs it but this looks like exactly what I was hoping for!

**Comment by xylar** [comment]
@anirban89, let's move the discussion on EKE to a separate issue.  I don't want this PR to be confused with an unrelated issue.

**Comment by xylar** [comment]
@lconlon, are you okay with this plot of the subpolar gyre?

image

We'd welcome feedback.  As I said, we can also tweak it to your liking later if this is okay as a first pass.

(It's duplicated twice in this plot just to test model vs. model comparison.  You can ignore that.)

**Comment by xylar** [comment]
@anirban89, would you have time to add a documentation page for this task?
Here's an example to work from:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/develop/docs/users_guide/tasks/climatologyMapEKE.rst
They're pretty formulaic but still I think it would be great if you take care of this rather than me since you put in the effort on this task.

**Comment by xylar** [comment]
You can make a new PR just for that.

**Pull request merged**
